huaweicloud/services/ims/resource_huaweicloud_ims_ecs_system_image.go

```diff
@@ -0,0 +1,582 @@
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			// The `description` field can be left blank, so the `Computed` attribute is not used.
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"max_ram": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"min_ram": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"tags": common.TagsSchema(),
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			// Attributes
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"visibility": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"image_size": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"disk_format": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"data_origin": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"os_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"active_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceEcsSystemImageCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg        = meta.(*config.Config)
+		region     = cfg.GetRegion(d)
+		createResp *cloudimages.JobResponse
+	)
+
+	client, err := cfg.ImageV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating IMS v2 client: %s", err)
+	}
+
+	imageTags := BuildCreateImageTagsParam(d)
+	createOpts := &cloudimages.CreateByServerOpts{
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		InstanceId:          d.Get("instance_id").(string),
+		MaxRam:              d.Get("max_ram").(int),
+		MinRam:              d.Get("min_ram").(int),
+		ImageTags:           imageTags,
+		EnterpriseProjectID: cfg.GetEnterpriseProjectID(d),
+	}
+	createResp, err = cloudimages.CreateImageByServer(client, createOpts).ExtractJobResponse()
+	if err != nil {
+		return diag.Errorf("error creating IMS image: %s", err)
+	}
+
+	imageId, err := WaitForCreateImageCompleted(client, d, createResp.JobID)
+	if err != nil {
+		return diag.Errorf("error waiting for IMS create ecs system image to complete: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error waiting for IMS create ECS system image to complete: %s", err)
```
Please check other places where similar issues exist.